src/snowflake/cli/plugins/nativeapp/teardown_processor.py

```diff
@@ -16,33 +16,55 @@
     CouldNotDropApplicationPackageWithVersions,
 )
 from snowflake.cli.plugins.nativeapp.manager import (
+    ApplicationObject,
     NativeAppCommandProcessor,
     NativeAppManager,
     ensure_correct_owner,
 )
-from snowflake.cli.plugins.nativeapp.utils import needs_confirmation
+from snowflake.cli.plugins.nativeapp.utils import (
+    needs_confirmation,
+)
 from snowflake.connector.cursor import DictCursor
 
 
 class NativeAppTeardownProcessor(NativeAppManager, NativeAppCommandProcessor):
     def __init__(self, project_definition: Dict, project_root: Path):
         super().__init__(project_definition, project_root)
 
-    def drop_generic_object(self, object_type: str, object_name: str, role: str):
+    def drop_generic_object(
+        self, object_type: str, object_name: str, role: str, cascade: bool = False
+    ):
         """
         Drop object using the given role.
         """
         with self.use_role(role):
             cc.step(f"Dropping {object_type} {object_name} now.")
             drop_query = f"drop {object_type} {object_name}"
+            if cascade:
+                drop_query += " cascade"
             try:
                 self._execute_query(drop_query)
             except:
                 raise SnowflakeSQLExecutionError(drop_query)
 
             cc.message(f"Dropped {object_type} {object_name} successfully.")
 
-    def drop_application(self, auto_yes: bool):
+    def _application_objects_to_str(
+        self, application_objects: ApplicationObject
+    ) -> str:
+        """
+        Returns a list in a "(Object Type) Object Name" format:
```

Comment:
Worth documenting here that database- and schema-level object names appear to be fully qualified.

Reply:
This comment seems fair and not yet resolved.